### PR TITLE
Golem shells require more dexterity

### PR DESCRIPTION
**Pull request by ParaGitV**
## What Does This PR Do

Currently there is nothing stopping borgs, drones, etc from finishing golem shells and having their own golem servants. This updates it so that humanoids are required to finish the golem.

## Why It's Good For The Game

Consistency to the idea that dexterous tasks require finer human manipulation.
Stops certain roles, such as maintenance drones, from completing golems and having their own servants when this shouldn't normally be possible.

## Testing

Confirmed that it worked as expected when testing humanoid mobs, silicon mobs, etc.

## Changelog
:cl:
tweak: Golem shells need to be finished by humanoids.
/:cl:

### CODE REVIEW

File: code/modules/awaymissions/mission_code/ghost_role_spawners/golems.dm [hunk 1]
```diff
@@ -42,6 +42,9 @@
 	if(istype(I, /obj/item/stack))
 		var/obj/item/stack/O = I
 		var/species = golem_shell_species_types[O.merge_type]
+		if(!istype(user, /mob/living/carbon/human))
```
Comment (Contrabang) [inline code/modules/awaymissions/mission_code/ghost_role_spawners/golems.dm:45]:
```suggestion
		if(!ishuman())
```

**Comment by Contrabang** [comment]
Also, this doesnt block golems from making more golems right? Free golems need to be able to do that, and preferably normal golems too. I think they are a carbon/human but i am not sure

**Comment by Contrabang** [comment]
Oh the changes I suggested made it fail CI, it should probably be `ishuman(user or whatever)`

File: code/modules/awaymissions/mission_code/ghost_role_spawners/golems.dm [hunk 2]
```diff
@@ -42,6 +42,9 @@
 	if(istype(I, /obj/item/stack))
 		var/obj/item/stack/O = I
 		var/species = golem_shell_species_types[O.merge_type]
+		if(!ishuman(user))
+			to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
+			return
```
Comment (Sirryan2002) [inline code/modules/awaymissions/mission_code/ghost_role_spawners/golems.dm:47]:
```suggestion
		if(!ishuman(user))
			to_chat(user, "<span class='warning'>You don't have the dexterity to do this!</span>")
			return
			
		var/obj/item/stack/O = I
		var/species = golem_shell_species_types[O.merge_type]
```
might aswell do this check before doing any other code here

**Review by Sirryan2002 — APPROVED** [review]
code looks good

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**